src/Components/CreateImageWizard/steps/Packages/Packages.tsx

```diff
@@ -959,22 +989,81 @@ const Packages = () => {
   const isGroupExpanded = (group: GroupWithRepositoryInfo['name']) =>
     expandedGroups.includes(group);
 
+  const [activeSortIndex, setActiveSortIndex] = useState<number>(0);
+  const [activeSortDirection, setActiveSortDirection] = useState<
+    'asc' | 'desc'
+  >('asc');
+
+  const getSortableRowValues = (
+    pkg: IBPackageWithRepositoryInfo
+  ): (string | number | ApiPackageSourcesResponse[] | undefined)[] => {
+    const { name, summary, stream, end_date, repository } = pkg;
+    return [name, summary, stream, end_date, repository];
+  };
+
+  let sortedPackages = transformedPackages;
+  sortedPackages = transformedPackages.sort((a, b) => {
+    const aValue = getSortableRowValues(a)[activeSortIndex];
+    const bValue = getSortableRowValues(b)[activeSortIndex];
+    if (typeof aValue === 'number') {
+      // Numeric sort
+      if (activeSortDirection === 'asc') {
+        return (aValue as number) - (bValue as number);
+      }
+      return (bValue as number) - (aValue as number);
+    } else {
+      // String sort
+      // handle packages with undefined stream
+      if (!aValue) {
+        return 1;
+      }
+      if (!bValue) {
+        return -1;
+      }
```

Comment:
This should solve a sorting bug when stream is undefined.

Reply:
Moved to previous PR: https://github.com/osbuild/image-builder-frontend/pull/3178